core/io/resource_loader.cpp

```diff
@@ -271,6 +270,22 @@ void ResourceLoader::_thread_load_function(void *p_userdata) {
 		}
 	}
 
+	// Post once for each waiting client who registered under lock; we know exactly
+	// how many there are.  The semaphore will already be allocated if there are any clients.
+	if (load_task.semaphore) {
+		for (int i = 0; i < load_task.semaphore_clients; i++) {
+			load_task.semaphore->post();
+		}
+	}
+
+#ifdef DEBUG_ENABLED
+	if (load_task.status == THREAD_LOAD_IN_PROGRESS) {
+		ERR_PRINT("Thread load task completed without reaching valid terminal state.  Broken implementation.");
```

Comment:
If this error should never be reached under normal use, it should be suggested that the user reports this issue:

```suggestion
		ERR_PRINT("Thread load task completed without reaching valid terminal state. Broken implementation (please report).");
```